### PR TITLE
:sparkles: Add e2e test for inventory-api

### PR DESCRIPTION
**Pull request by yanmxa**
## Summary

1. Kraft Compatibility: Installation error fixed in this PR: https://github.com/stolostron/multicluster-global-hub/pull/1232, visit the discussion https://github.com/orgs/strimzi/discussions/10836 for more detail
2. Inventory API Accession: https://github.com/project-kessel/inventory-api/issues/243
3. Inventory data conflicts: https://github.com/project-kessel/inventory-api/pull/244 
4. The defined [kafka event](https://github.com/project-kessel/docs/blob/main/src/content/docs/inventory/kafka-event.md) is different from the eventing API(the [relationship_data](https://github.com/project-kessel/inventory-api/blob/main/internal/eventing/api/event.go#L32) is not exist)!
5. The kafka deletion event of the relationship is not different from the [definition of the Kafka event](https://github.com/project-kessel/inventory-api/blob/main/internal/eventing/api/event.go#L32). 
Relationship example:
- Create
```
Context Attributes,
  specversion: 1.0
  type: redhat.inventory.resources_relationship.k8s-policy_ispropagatedto_k8s-cluster.created
  source: https:
  subject: /resources_relationship/k8s-policy_ispropagatedto_k8s-cluster/policy-id-24546
  id: 4153de06-a366-11ef-8785-8a12bc3224a5
  time: 2024-11-15T15:28:24.975603472Z
  datacontenttype: application/json
Data,
  {
    "metadata": {
      "id": 12,
      "last_reported": "2024-11-15T15:28:24.975603472Z",
      "relationship_type": "k8s-policy_ispropagatedto_k8s-cluster"
    },
    "reporter_data": {
      "reporter_instance_id": "guest",
      "reporter_type": "ACM",
      "last_reported": "2024-11-15T15:28:24.976340254Z",
      "subject_local_resource_id": "policy-id-24546",
      "object_local_resource_id": "cluster-id-51474",
      "reporter_version": "2.12"
    },
    "relationship_data": {
      "status": "VIOLATIONS",
      "k8s_policy_id": 0,
      "k8s_cluster_id": 0
    }
  }
```
- Update
```
Context Attributes,
  specversion: 1.0
  type: redhat.inventory.resources_relationship.k8s-policy_ispropagatedto_k8s-cluster.updated
  source: https:
  subject: /resources_relationship/k8s-policy_ispropagatedto_k8s-cluster/policy-id-24546
  id: 43b6eeba-a366-11ef-8785-8a12bc3224a5
  time: 2024-11-15T15:28:28.982812429Z
  datacontenttype: application/json
Data,
  {
    "metadata": {
      "id": 0,
      "last_reported": "2024-11-15T15:28:28.982812429Z",
      "relationship_type": "k8s-policy_ispropagatedto_k8s-cluster"
    },
    "reporter_data": {
      "reporter_instance_id": "guest",
      "reporter_type": "ACM",
      "last_reported": "2024-11-15T15:28:28.982812429Z",
      "subject_local_resource_id": "policy-id-24546",
      "object_local_resource_id": "cluster-id-51474",
      "reporter_version": "2.12"
    },
    "relationship_data": {
      "status": "NO_VIOLATIONS",
      "k8s_policy_id": 0,
      "k8s_cluster_id": 0
    }
  }
```
- Delete
```
Context Attributes,
  specversion: 1.0
  type: redhat.inventory.resources_relationship.k8s-policy_ispropagatedto_k8s-cluster.deleted
  source: https:
  subject: /resources_relationship/k8s-policy_ispropagatedto_k8s-cluster/cluster-id-51474_policy-id-24546
  id: 444fd42d-a366-11ef-8785-8a12bc3224a5
  time: 2024-11-15T15:28:29.984868624Z
  datacontenttype: application/json
Data,
  {
    "metadata": {
      "id": 0,
      "last_reported": "0001-01-01T00:00:00Z",
      "relationship_type": ""
    },
    "reporter_data": {
      "reporter_instance_id": "guest",
      "reporter_type": "",
      "last_reported": "0001-01-01T00:00:00Z",
      "subject_local_resource_id": "local-id",
      "object_local_resource_id": "local-id",
      "reporter_version": "1.0"
    }
  }
```

## Related issue(s)

Fixes # https://issues.redhat.com/browse/ACM-15034

## Tests
* [ ] Unit/function tests have been added and incorporated into `make unit-tests`.
* [ ] Integration tests have been added and incorporated into `make integration-test`.
* [x] E2E tests have been added and incorporated into `make e2e-test-all`.
* [ ] List other manual tests you have done.

### CODE REVIEW

File: test/script/config/.gitkeep
Comment (clyang82) [inline test/script/config/.gitkeep:1]: do you need this file?

Comment (yanmxa) [reply test/script/config/.gitkeep:1]: Remove it

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**